tango/integrations/wandb/flax_train_callback.py

```diff
@@ -0,0 +1,172 @@
+from typing import Any, Dict, List, Optional
+
+import wandb
+
+from tango.common.exceptions import ConfigurationError
+from tango.integrations.flax.train_callback import TrainCallback
+
+from .workspace import WandbWorkspace
+
+
+@TrainCallback.register("wandb::log_flax")
+class WandbTrainCallback(TrainCallback):
```

Comment:
Other than the base `TrainCallback`, is this entirely the same as the torch wandb callback?

Reply:
The torch callback uses the `TrainingEngine` to access some variables whereas there is no training engine in flax. It also uses some torch utility functions to monitor gpu usage.